neuronpp/utils/record.py

```diff
@@ -143,17 +151,66 @@ def _plot_animate(self, steps=10000, y_lim=None, position=None):
 
                 ax.set_xlim(t.min(), t.max())
                 if y_lim is None:
-                    ax.set_ylim(r.min()-(np.abs(r.min()*0.05)), r.max()+(np.abs(r.max()*0.05)))
+                    y_limits = (r.min() - (np.abs(r.min() * 0.05)), r.max() + (np.abs(r.max() * 0.05)))
+                    ax.set_ylim(y_limits)
 
                 # update data
                 line.set_data(t, r)
-
+                if show_true_predicted:
+                    # info draw triangles for true and predicted classes
+                    if true_labels is not None:
+                        true_x, pred_x = self._true_predicted_class_marks(label=true_labels[i], true_class=true_class,
+                                                                          pred_class=pred_class, t=t, r=r,
+                                                                          stepsize=stepsize, dt=dt)
+                    else:
+                        raise ValueError("True_labels parameter need to be given if show_true_prediction is True")
+                    if y_lim is None:
+                        true_y = [y_limits[0] + np.abs(y_limits[0]) * 0.09] * len(true_x)
+                        pred_y = [y_limits[1] - np.abs(y_limits[1] * 0.11)] * len(pred_x)
```

Comment:
variables  `y_lim`  and  `y_limits` are too confusing. It should be a single variable. I think you should override `y_lim` in line 154.

Reply:
Actually no, because the parameter y_lim is global for all plots (a tuple in parameters); while y_limits are compoted per individual plots. During the course of simulation, the activations tend to have different values.
Changed the name y_limits to this_plot_y_limits.

Reply:
Actually this shall never work as it was before and is now: if the max--min values come too near, the y_lim plots is much different from what we can read from data. There must be a method which shall first compute the wanted limits, then set them, then read what matplotlib really used. Is there one?